### ISSUE TITLE
Add support for higher resolution templates

### ISSUE THREAD

**Issue by zonca** [simonsobs/so_pysm_models#16]
After we have the gaussian foreground simulations done, create a repository `so_pysm_models_data` and import high resolution templates from https://github.com/NicolettaK/PySM_public
Then in this repository create classes for those models that download the high resolution templates on-demand.
This is the same system implemented in PySM 3:

* we put templates in https://github.com/healpy/pysm-data
* they are then available in https://healpy.github.io/pysm-data/
* downloaded on-demand with `astropy.io.data.get_pkg_data_filename`

**Comment by zonca** [comment]
@NicolettaK here is the plan to add your new templates for PySM, I'll start working on this after we have completed the first simulations with gaussian foregrounds

**Comment by zonca** [comment]
actually Github is not suitable for 4096 maps, use NERSC web hosting:

Use `/global/project/projectdirs/cmb/www/so_pysm_models_data` which is published to http://portal.nersc.gov/project/cmb/so_pysm_models_data/index.html

**Comment by NicolettaK** [comment]
@zonca I am ready to copy a first version of high resolution templates to the directory you mentioned, but I don't have permission to write on it, how should we proceed?

**Comment by zonca** [comment]
@NicolettaK please copy them to your scratch and make them readable by the `sobs` group

**Comment by NicolettaK** [comment]
@zonca I've copied here at nersc the high resolution maps, as well as the old PySM templates:
`/global/cscratch1/sd/krach/SObs/FGawg/so_template`
You should have access to read them.

If you copy them on the `so_pysm_models_data` directory I can start working on the classes to download and use them

**Comment by zonca** [comment]
ok done

**Comment by zonca** [comment]
A suggestion on how to implement this:

normally you use PySM with:

```
sky_config = {
    'synchrotron' : models("s1", nside),
    'dust' : models("d1", nside),
    'freefree' : models("f1", nside),
    'cmb' : models("c1", nside),
    'ame' : models("a1", nside),
}
```

I think you can implement `so_pysm_models.get_so_model` function very similar to the function `models` inside `nominal.py` of PySM. This should get the same keys of PySM, e.g. `d1` but instead of reading the PySM templates, it reads the SO templates.

You can use this snippet to download templates when needed and then cache them:

https://github.com/healpy/pysm/blob/master/pysm/component_models/template.py#L256

If you get stuck anywhere along this, please open a pull request and ask for feedback.

**Comment by zonca** [comment]
See #20 and #23

**Comment by zonca** [comment]
@NicolettaK ok, I think the code is fine, only documentation is missing, i.e. some docstrings and then detailed description in `models.rst`.

I also implemented that if you are running at NERSC, it automatically picks up the local files from `/project`.
If not, it downloads them on the fly and stores them in a cache in the home:

```
from astropy.utils import data
ls -lah /home/zonca/.astropy/cache/download/py3/
total 2.4G
drwxrwxr-x 2 zonca zonca 4.0K Mar 12 21:08 ./
drwxrwxr-x 4 zonca zonca   26 May 23  2018 ../
-rw------- 1 zonca zonca  13M Mar 12 21:07 1458b329fee938b45b51371aad09814d
-rw------- 1 zonca zonca  13M Mar 12 21:08 1a0cfdfa469c5c44632b8c0343ff60df
-rw------- 1 zonca zonca  13M Mar 12 21:07 1f3e10d1e8a63f85839ef3cfd4839db0
-rw------- 1 zonca zonca  13M Mar 12 21:07 36e1fc4b8f3a0bc4cfa923c9f2cf2cef
-rw------- 1 zonca zonca 769M Mar 12 21:05 6fa6cc74c1cb92f022bd3c1cc12974fb
-rw------- 1 zonca zonca  49K Mar 12 21:08 850a3b248c7e504650ca5027f7fa3227
-rw------- 1 zonca zonca  13M Mar 12 21:08 93cafe1496e997f48c628daabda836f9
-rw------- 1 zonca zonca 769M Mar 12 21:05 c250fc4311decb1e312cd11f797b24c5
-rw------- 1 zonca zonca  13M Mar 12 21:07 da64b8daa8607508c46e3c60d802a65b
-rw------- 1 zonca zonca 769M Mar 12 21:05 fce5583c699ae58cf244fc321cea3d1b
-rw-rw-r-- 1 zonca zonca  886 Mar 12 21:08 urlmap.bak
-rw-rw-r-- 1 zonca zonca 4.6K Mar 12 21:08 urlmap.dat
-rw-rw-r-- 1 zonca zonca  886 Mar 12 21:08 urlmap.dir

data.get_cached_urls()
['http://portal.nersc.gov/project/cmb/so_pysm_models_data/dust_T_ns4096.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/dust_Q_ns4096.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/dust_U_ns4096.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/synch_T_ns512.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/synch_Q_ns512.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/synch_U_ns512.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/freefree_T_ns512.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/ame_T_ns512.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/ame2_T_ns512.fits',
 'http://portal.nersc.gov/project/cmb/so_pysm_models_data/ame_emissivity.txt']
```

**Comment by zonca** [comment]
implemented in #23
